### PR TITLE
Core: Implement FanoutPositionDeleteWriter

### PR DESCRIPTION
**Pull request by marton-bod**
Implement new `FanoutPositionDeleteWriter`, similar to the already-existing `FanoutDataWriter` class. 
This new DeleteWriter implementation should provide an alternative to `ClusteredPositionDeleteWriter`  for those users who are not inserting their data clustered by partition spec/partition values.

@aokolnychyi @openinx @pvary - can you please take a look? Thanks!

### CODE REVIEW

**Comment by rdblue** [comment]
@marton-bod, I would rather not include this if we don't know that it is definitely needed. Otherwise it would be easy to start using it even though there are better ways to prepare the data so that it isn't needed. Does that make sense? If there's a way around adding a class, then we should avoid adding it.

**Comment by aokolnychyi** [comment]
Fanout writers may be inefficient so I'd rather use `_spec_id` and `_partition` metadata columns if possible. I'd be happy to review PRs to make those columns available in query engines other than Spark.

That being said, we should probably still enable ORC tests in `TestPartitioningWriters`.

File: core/src/main/java/org/apache/iceberg/io/FanoutPositionDeleteWriter.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    this.fileFactory = fileFactory;
+    this.io = io;
+    this.targetFileSizeInBytes = targetFileSizeInBytes;
+    this.deleteFiles = Lists.newArrayList();
+    this.referencedDataFiles = CharSequenceSet.empty();
+  }
+
+  @Override
+  protected FileWriter<PositionDelete<T>, DeleteWriteResult> newWriter(PartitionSpec spec, StructLike partition) {
+    return new RollingPositionDeleteWriter<>(writerFactory, fileFactory, io, targetFileSizeInBytes, spec, partition);
```
[the remark this answers is not in the document]
Comment (aokolnychyi) [reply core/src/main/java/org/apache/iceberg/io/FanoutPositionDeleteWriter.java:55]: What about enabling them in a separate PR?

**Comment by marton-bod** [comment]
@rdblue @aokolnychyi Makes sense! Thanks for reviewing, I'll close this PR.
I'll open a new PR for enabling ORC tests in `TestPartitioningWriters`.

**Pull request closed without merging**